### PR TITLE
chore: [IOPID-1015] Biometric refinement

### PR DESCRIPTION
**Pull request by sabontech**
## Short description
This pr adds the logic that allows the app not to show the biometric activation request if it has already been activated in a previous login and adds, in the case of FACE_ID, the prompt requesting permissions upon activation.

## List of changes proposed in this pull request
- ts/sagas/startup/onboarding/biometric/checkAcknowledgedFingerprintSaga.ts: added check to see if it had been activated in the last login.
- [ts/screens/onboarding/biometric&securityChecks/FingerprintScreen.tsx , ts/screens/profile/SecurityScreen.tsx] : added permission request prompt in case of faceid.
- ts/store/reducers/index.ts: biometric preference persisted to get it also after the end of a session.
- ts/store/reducers/onboarding.ts: added reset of biometric control also after session expired.
- ts/store/reducers/persistedPreferences.ts: added cross session control.
- ts/utils/biometrics.ts: added function that handles the biometric activation.

## How to test
With a new installation, try activating biometrics during onboarding:
- If the biometric is not FaceID, it should activate (or not) the feature without prompting.
- If the biometric is FaceID, when the active button is pressed, it should present the request for permissions:
    - If you give permissions and the authentication fails, it should not proceed.
    - If you give permissions and the authentication is successful, it should proceed and enable the feature.
    - If you don't give permissions, it should proceed and not activate the feature.
    
After logout or session expired, if the biometric has been activated during last session, it should not ask you for activation again after login, unless this is done with a different account than the previous one.

### CODE REVIEW

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]
I suggest to make some tests for the new function `mayUserActivateBiometric`.

File: ts/utils/biometrics.ts [hunk 1]
```diff
@@ -121,3 +122,41 @@ export const isDevicePinSet = (): Promise<boolean> =>
...
+              ),
+              fallbackEnabled: false
+            } as AuthenticateIOS)
+              .then(_ => resolve("ACTIVATED"))
+              .catch((err: Errors) => {
+                reject(handleErrorDuringBiometricActivation(err));
+              });
+          } else {
+            resolve("ACTIVATED");
+          }
```
[the remark this answers is not in the document]
Comment (sabontech) [reply ts/utils/biometrics.ts:150]: For this specific case, although it's possible from the code, it's practically impossible for "UNAVAILABLE" to be returned at that point. 
You can get to the authentication activation screen (where this function is used) only if a type of biometric authentication has been recognized as valid and available. 
However, in case it becomes "UNAVAILABLE" for some reason, after reaching the authentication activation screen, I thought it would still be useful to let the user activate the biometric (as happened so far). 
The only annoyance the user will receive, will be the system permission screen (only in case of FaceID) on first use.
In short, this is an extremely rare case and I think it is more correct to activate everything rather than launch an error.

File: ts/utils/biometrics.ts [hunk 2]
```diff
@@ -121,3 +122,41 @@ export const isDevicePinSet = (): Promise<boolean> =>
...
+        })
+        .catch(_ => reject("SENSOR_ERROR"));
+    });
```
[the remark this answers is not in the document]
Comment (sabontech) [reply ts/utils/biometrics.ts:153]: I agree, it's always a good idea to write some tests 🙂

File: ts/utils/biometrics.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sabontech) [reply ts/utils/biometrics.ts:150]: If you're thinking of a user who gets to the biometric activation screen and removes permission from the system settings, simply whatever choices he makes, will be irrelevant and the app won't consider biometrics from that point on. Furthermore, when it comes to a clean installation, it would not be possible to disable the setting from the system as it is not available at that point of onboarding yet.

[the remark this answers is not in the document]
Comment (shadowsheep1) [reply ts/utils/biometrics.ts:150]: Okay then. Let's add some tests and their explanations too! 🚀

File: ts/utils/__tests__/biometrics.test.ts [hunk 3]
```diff
@@ -86,3 +88,151 @@ describe("biometricAuthenticationRequest function", () => {
...
+    try {
+      await Biometric.biometricFunctionForTests.mayUserActivateBiometricWithDependency(
+        getBiometricsTypeRejectedMock
+      );
+    } catch (error) {
+      expect(error).toBe("SENSOR_ERROR");
+    }
+  });
+
+  it("returns ACTIVATED when getBiometricsType promise resolves FACE_ID and the authentication  is successful", async () => {
```
Comment (shadowsheep1) [inline ts/utils/__tests__/biometrics.test.ts:117]:
```suggestion
  it("returns ACTIVATED when getBiometricsType promise resolves FACE_ID and the authentication is successful", async () => {
```

**Review by shadowsheep1 — APPROVED** [review]
Apart from a minor changes, LGTM 👍
